edalize/symbiflow.py

```diff
@@ -0,0 +1,160 @@
+import logging
+import os.path
+import platform
+import re
+import subprocess
+
+from edalize.edatool import Edatool
+from edalize.yosys import Yosys
+from importlib import import_module
+
+logger = logging.getLogger(__name__)
+
+""" Symbiflow backtend
+
+A core (usually the system core) can add the following files:
+
+- Standard design sources (Verilog only)
+
+- Constraints: unmanaged constraints with file_type SDC, pin_constraints with file_type PCF and placement constraints with file_type xdc
+
+"""
+
+
+class Symbiflow(Edatool):
+
+    argtypes = ["vlogdefine", "vlogparam", "generic"]
+
+    @classmethod
+    def get_doc(cls, api_ver):
+        if api_ver == 0:
+            symbiflow_help = {
+                "members": [
+                    {
+                        "name": "package",
+                        "type": "String",
+                        "desc": "FPGA chip package (e.g. clg400-1)",
+                    },
+                    {
+                        "name": "part",
+                        "type": "String",
+                        "desc": "FPGA part type (e.g. xc7a50t)",
+                    },
+                    {
+                        "name": "builddir",
+                        "type": "String",
+                        "desc": 'directory where all the intermediate files will be stored (default "build")',
+                    },
+                    {
+                        "name": "vendor",
+                        "type": "String",
+                        "desc": 'Target architecture. Currently only "xilinx" is supported',
+                    },
+                    {
+                        "name": "pnr",
+                        "type": "String",
+                        "desc": 'Place and Route tool. Currently only "vpr" is supported',
+                    },
+                    {
+                        "name": "options",
+                        "type": "String",
+                        "desc": "Tool options. If not used, default options for the tool will be used",
+                    },
+                    {
+                        "name" : "seed",
```

Comment:
Do we really need to expose seed as a separate option. Looking at the Makefile template below I see that all other options are sent to symbiflow_pack and symbiflow_place, symbiflow_route and symbiflow_write_fasm. Is it ok to pass --seed to all stages as well, or is that one special?

Reply:
No, we do not need to expose ``seed`` as a separate option. I updated toolchain wrappers to allow passing additional vpr options (that are added to the default one from the wrapper). (https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1682)

Yes, we can set ``seed`` option in all vpr stages, but ``seed`` is only used in ``place``. I changed seed to be part of ``additional_vpr_options`` and added this to all stages of vpr.